trie-db/src/node.rs

```diff
@@ -40,21 +41,45 @@ pub fn decode_hash<H: Hasher>(data: &[u8]) -> Option<H::Out> {
 	Some(hash)
 }
 
+/// Value representation in `Node`.
+#[derive(Eq, PartialEq, Clone)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub enum Value<'a> {
+	/// Value byte slice.
+	Value(&'a [u8]),
+	/// Hash byte slice and the actual value if accessed.
+	HashedValue(&'a [u8], Option<DBValue>),
+}
+
+impl<'a> Value<'a> {
+	pub(crate) fn new(value: &'a [u8], threshold: Option<u32>) -> Option<Self> {
```

Comment:
maybe we could improve the name here, like Value::Value could be renamed Value::Inlined, and Value::HashedValue could be renamed Value::ValueNode,

and `new` which only tries to create a `Value::Value` could be renamed `new_inlined`.

That said I'm not sure of the name I proposed either. so maybe better to keep as it is